apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2.java

```diff
@@ -1098,4 +1105,138 @@ Timer.Context createTimerContext(String partId) {
     return CommonTransformerUtils.createMetricsTimer(
         metricRegistry, getClass().getSimpleName(), "transform", partId);
   }
+
+  /**
+   * concatenates Part C Contract And Pbp.
+   *
+   * @param beneficiary the value for {@link Beneficiary}
+   * @return concatenated Contract And Pbp
+   */
+  private String concatenatePartCContractAndPbp(Beneficiary beneficiary) {
+    int month = Calendar.getInstance().get(Calendar.MONTH);
+    String contractId = getPartCContractNumber(beneficiary, month).orElse(null);
+    String pbpId = getPartCPbpNumber(beneficiary, month).orElse(null);
+    if (contractId != null && pbpId != null) {
+      return contractId + "-" + pbpId;
+    }
+    return null;
+  }
+
+  /**
+   * concatenates Part D Contract And Pbp.
+   *
+   * @param beneficiary the value for {@link Beneficiary}
+   * @return concatenated Contract And Pbp
+   */
+  private String concatenatePartDContractAndPbp(Beneficiary beneficiary) {
+
+    int month = Calendar.getInstance().get(Calendar.MONTH);
```

Comment:
Might be nice to make a helper function for this rather than duplicating the logic for part C and part D